core/src/storage/cache.py

```diff
@@ -184,12 +207,30 @@ def set(key: int, value: bytes) -> None:
     _SESSIONS[_active_session_idx].set(key, value)
 
 
-def get(key: int) -> bytes:
+@overload
+def get(key: int) -> bytes | None:
+    ...
+
+
+@overload
+def get(key: int, default: T) -> bytes | T:  # noqa: F811
+    ...
+
+
+def get(key: int, default: T | None = None) -> bytes | T | None:  # noqa: F811
```

Comment:
weirdly enough I can't: the overload signatures are used at call site, and this signature is used to evaluate the function body.

note that all this overload business is only necessary because `get(key: int, default: T = None) -> bytes | T` is not understood by mypy